OpenRA.Platforms.Default/OpenGL.cs

```diff
@@ -553,5 +662,61 @@ public static void WriteGraphicsLog(string message)
 			for (var i = 0; i < extensionCount; i++)
 				Log.Write("graphics", glGetStringi(GL_EXTENSIONS, (uint)i));
 		}
+
+		public static void CheckGLError()
+		{
+			// Let the debug message handler log the errors instead.
+			if (Features.HasFlag(GLFeatures.DebugMessagesCallback))
+				return;
+
+			var type = glGetError();
+			if (type == GL_NO_ERROR)
+				return;
+
+			string errorText;
+			errorText = ErrorToText.TryGetValue(type, out errorText) ? errorText : type.ToString("X");
+			var error = "GL Error: {0}\n{1}".F(errorText, new StackTrace());
+
+			WriteGraphicsLog(error);
+
+			const string exceptionMessage = "OpenGL Error: See graphics.log for details.";
+
+			if (type == GL_OUT_OF_MEMORY)
+				throw new OutOfMemoryException(exceptionMessage);
+
+			throw new InvalidOperationException(exceptionMessage);
+		}
+
+		static void DebugMessageHandler(int source, int type, uint id, int severity, int length, StringBuilder message, IntPtr userparam)
+		{
+			string error;
```

Comment:
What is this variable for? As far as I can see it's only used directly and once after being written to.

Reply:
You can't define the same variable in different branches of a switch statement. I opted to build the error messages on one line and then throw/log/write on another for readability and maintainability.

Reply:
What I wanted to say was that I wouldn't define a variable at all, as it doesn't seem to be used more than once anyway(?).

Reply:
I can certainly inline the usage if you insist, my argument against is that `Nesting(Methods(), Like(), This())` makes for code that is harder to read and maintain.

Reply:
I would disagree for this particular instance since it should be pretty clear what the code is doing regardless (especially since the method is named appropriately) but don't really care either way. So I guess you can leave it as is unless someone else wants it changed too.